### ISSUE TITLE
use deployments and not replicacontrollers

### ISSUE THREAD

**Issue by MHBauer** [IBM/guestbook#8]
from the docs:
```NOTE: A Deployment that configures a ReplicaSet is now the recommended way to set up replication.```

**Comment by duglin** [comment]
@linsun any objection to this change?

**Comment by linsun** [comment]
@duglin no objection but would prefer only in v2 of the service... so we can keep v1 to be vanilla from the original guestbook.

**Comment by MHBauer** [comment]
Why? What educational purpose does it serve?

Where is the original?

**Comment by cmluciano** [comment]
+1 to switching to deployments, it's often confusing to say "forget everything I just said about replicasets and now let's use deployments"

**Comment by linsun** [comment]
original is from k8s repo.  the purpose is to show people you can take the sample as it is with very minimum change and deploy within istio.

**Comment by MHBauer** [comment]
Well, now it's our sample code, so let's make it use best practices. Everything possible with replicacontrollers should be possible with deployments.

**Comment by linsun** [comment]
right, that is why i'm all for the change in v2.  for v1, the focus is to show people they can bring services onto istio easily with very few changes.  this shows people starts with something they have it working (which maybe designed for slightly older version of k8s) but users can modernize it in v2.

**Comment by MHBauer** [comment]
I'm not going to teach people that for a workshop. Maybe we need a vLegacy as well.

**Comment by linsun** [comment]
for the record, I'm ok with changing it in v1 and v2 to keep it simple given many of you shared concerns on using RCs.

**Comment by frankbu** [comment]
I absolutely agree it should be changed to Deployment in all versions. Nobody cares about RCs anymore. Anybody using them these days is misguided.

**Comment by MHBauer** [comment]
#dibs
